### PR TITLE
bug fix to allow CI to return fakes for lowPU mode

### PR DESCRIPTION
**Pull request by isabellavesely**
When running setupCombine with root files, for lowPU modes (where PhotonInduced is not a valid process), CardTool.py skipped the addition of fake impacts. This change updates CardTool.py to retain the fake impact information regardless of setupCombine working under root or hdf5 modes.

Furthermore, there is an update (need to test) to the CI that switches lowPU fit run to operate under combined hdf5 mode.

### CODE REVIEW

File: wremnants/CardTool.py [hunk 1]
```diff
@@ -1050,9 +1047,11 @@ def writeLnNSystematics(self):
         # exit this function when a syst is applied to no process (can happen when some are excluded)
         for name,info in self.lnNSystematics.items():
             if self.isExcludedNuisance(name): continue
+            # lowPU does not include PhotonInduced as a process. skip it:
+            if ("lowpu" in self.datagroups.mode) and (name == 'CMS_PhotonInduced'): continue
```
Comment (davidwalter2) [inline wremnants/CardTool.py:1051]: In my opinion this should go into setupCombine instead, which means skipping this line for low PU: https://github.com/WMass/WRemnants/blob/main/scripts/combine/setupCombine.py#L748

Comment (isabellavesely) [reply wremnants/CardTool.py:1051]: Yeah, this makes more sense, I'll update this -- thanks.

**Comment by davidwalter2** [comment]
Are the changes in low PU as expected?
this PR:
https://cmsmwbot.web.cern.ch/WMassAnalysis/PRValidation/PR492/2024_07_04/lowPU/
Reference:
https://cmsmwbot.web.cern.ch/WMassAnalysis/PRValidation/ReferenceRuns/2024_07_04_4c06160/lowPU/
If so, can it be merged?

**Comment by isabellavesely** [comment]
> Are the changes in low PU as expected? 
this PR: https://cmsmwbot.web.cern.ch/WMassAnalysis/PRValidation/PR492/2024_07_04/lowPU/ 
Reference: https://cmsmwbot.web.cern.ch/WMassAnalysis/PRValidation/ReferenceRuns/2024_07_04_4c06160/lowPU/ 
If so, can it be merged?

Yes, I did expect this difference in the new CI output. In the reference, pre-PR, the CI didn't incorporate fakes into any lowPU impact results at all. After the change, the results from the CI are nearly identical to the results I found manually (in which I used the same input files as the CI, but avoided the root/txt file method and instead used setupCombine directly through hdf5 with both inputs). See here, where left is mine and right is after PR:
image

image

But there are some small numerical differences in the above. I was planning to review/re-run my manual version, and check that the input files are the most updated before merging, unless a merge already seems reasonable. Let me know if there's any immediate feedback.

**Comment by davidwalter2** [comment]
It looks good to me but we can also wait with the merge, as you like

**Comment by isabellavesely** [comment]
Ok, re-ran myself to check and got the results I expected, so I'm more confident that we are good to go with the merge.

**Pull request merged**